### PR TITLE
fixed updating dynamic years when min and max are chosen

### PR DESCRIPTION
**Pull request by victoriaaa234**
waiting for @mmalerba's PR that fixes active date functionality

### CODE REVIEW

**Comment by raygervais** [comment]
I am alright with my commits being used in this PR. It's awesome that you took this on @victoriaaa234!

File: src/lib/datepicker/multi-year-view.ts [hunk 1]
```diff
@@ -131,7 +134,21 @@ export class MatMultiYearView<D> implements AfterContentInit {
...
     this._todayYear = this._dateAdapter.getYear(this._dateAdapter.today());
     let activeYear = this._dateAdapter.getYear(this._activeDate);
-    let activeOffset = activeYear % yearsPerPage;
+    // Default Behavior for Offset
+    let activeOffset = yearsPerPage / 2;
+    if (!this._maxDate && this._minDate) {
+      activeOffset = 0;
+    }
+
+    if (this._maxDate) {
```
Comment (tinayuangao) [inline src/lib/datepicker/multi-year-view.ts]: Add some comments to explain how `activeOffset` is calculated? Add tests?

File: src/lib/datepicker/multi-year-view.ts [hunk 2]
```diff
@@ -101,6 +101,9 @@ export class MatMultiYearView<D> implements AfterContentInit {
   /** Emits the selected year. This doesn't imply a change on the selected date */
   @Output() readonly yearSelected: EventEmitter<D> = new EventEmitter<D>();
 
+  /** Emits when any date is activated. */
+  @Output() readonly activeDateChange: EventEmitter<D> = new EventEmitter<D>();
```
Comment (tinayuangao) [inline src/lib/datepicker/multi-year-view.ts]: Add test for `activeDateChange`?

**Comment by j-ramirez-bi4group** [comment]
+1

**Comment by migosantos** [comment]
any updates on when this feature will be released?

**Pull request closed without merging**